### PR TITLE
feat: include feedId to content_preference pk

### PR DESCRIPTION
**Pull request by capJavert**
We need feedId as well as PK, [thread](https://dailydotdev.slack.com/archives/C07ATUPBPJ9/p1731420025841499)

- updated PK and column
- removed `feed*` types (not needed anymore since feedId will make them unique together with their original type)
- will update prod data before merging so constraints can be created

### CODE REVIEW

File: src/migration/1731424255967-ContentPreferenceReplicaFull.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `ALTER TABLE "content_preference" REPLICA IDENTITY FULL`,
+    );
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `ALTER TABLE "content_preference" REPLICA IDENTITY DEFAULT`,
+    );
```
Comment (capJavert) [inline src/migration/1731424255967-ContentPreferenceReplicaFull.ts:17]: Needed for clickhouse sync

File: src/common/contentPreference.ts [hunk 2]
```diff
@@ -114,6 +112,7 @@ const followUser: FollowEntity = async ({ ctx, id, status }) => {
 
     const contentPreference = repository.create({
       userId: ctx.userId,
+      feedId: ctx.userId,
```
Comment (capJavert) [inline src/common/contentPreference.ts:115]: by default feedId is userId, for custom feeds its gonna be actual id of the feed

File: __tests__/contentPreference.ts [hunk 3]
```diff
@@ -103,32 +103,45 @@ describe('query userFollowers', () => {
...
+        userId: `${item.id}-ufq`,
+      })),
+    );
+
     const now = new Date();
 
     await con.getRepository(ContentPreferenceUser).save([
       {
         userId: '2-ufq',
+        feedId: '2-ufq',
```
Comment (rebelchris) [inline __tests__/contentPreference.ts:120]: How did this test not fail before 😢

Comment (rebelchris) [reply __tests__/contentPreference.ts:120]: But maybe we should add 1 user - 3 feeds with same tags for safety

File: src/common/contentPreference.ts (hunk 2 above)
Comment (rebelchris) [reply src/common/contentPreference.ts:115]: Yeah internally userid is just yet another feed, i like it 👍

File: src/common/contentPreference.ts [hunk 4]
```diff
@@ -247,7 +246,7 @@ const unfollowSource: UnFollowEntity = async ({ ctx, id }) => {
 
     await repository.delete({
       userId: ctx.userId,
-      sourceId: id,
```
Comment (rebelchris) [inline src/common/contentPreference.ts:250]: Dont you need these still?

**Review by rebelchris — APPROVED** [review]
Checked from phone, is the deletion needed, dont we need the sync still else we might delete to globally?

Also maybe add what would previously fail as a test?

File: src/common/contentPreference.ts (hunk 4 above)
Comment (capJavert) [reply src/common/contentPreference.ts:250]: no, it was a mistake from before, referenceId is sourceId in this case and also PK so delete is faster

Comment (rebelchris) [reply src/common/contentPreference.ts:250]: Ah but whats the denominator between source-tag then?

Matbe im not seeing it on the changeset though

Comment (capJavert) [reply src/common/contentPreference.ts:250]: Ah that, typeorm automatically adds `type` where clause because I am using `ContentPreferenceSource` entity

Comment (rebelchris) [reply src/common/contentPreference.ts:250]: Ah i see, that answers it 🙏

File: __tests__/contentPreference.ts (hunk 3 above)
Comment (capJavert) [reply __tests__/contentPreference.ts:120]: Added in 2815f85c2e0863a391f38b3d0f34048a3c75ef8c

**Review by rebelchris — APPROVED** [review]

**Pull request merged**